src/samples/coap/lwm2m-client.c

```diff
@@ -281,23 +330,178 @@ read_security_server_obj(void *instance_data, void *user_data,
+    }
+
+    if (r >= 0)
+        printf("Resource written to Security object at /0/%" PRIu16 "/%" PRIu16 "\n",
+            instance_id, res->id);
+    return r;
+}
+
+static int
+write_security_tlv(void *instance_data, void *user_data,
+    struct sol_lwm2m_client *client,
+    uint16_t instance_id, struct sol_vector *tlvs)
+{
+    int r;
+    uint16_t i;
+    struct sol_lwm2m_tlv *tlv;
+    struct security_obj_instance_ctx *instance_ctx = instance_data;
+
+    SOL_VECTOR_FOREACH_IDX (tlvs, tlv, i) {
+        SOL_BUFFER_DECLARE_STATIC(buf, 64);
+
+        switch (tlv->id) {
+        case SECURITY_SERVER_SERVER_URI_RES_ID:
+            r = sol_lwm2m_tlv_get_bytes(tlv, &buf);
+            sol_blob_unref(instance_ctx->server_uri);
+            instance_ctx->server_uri = sol_buffer_to_blob(&buf);
+            break;
+        case SECURITY_SERVER_IS_BOOTSTRAP_RES_ID:
+            r = sol_lwm2m_tlv_get_bool(tlv, &instance_ctx->is_bootstrap);
+            break;
+        case SECURITY_SERVER_SERVER_ID_RES_ID:
+            r = sol_lwm2m_tlv_get_int(tlv, &instance_ctx->server_id);
+            break;
+        case SECURITY_SERVER_CLIENT_HOLD_OFF_TIME_RES_ID:
+            r = sol_lwm2m_tlv_get_int(tlv, &instance_ctx->client_hold_off_time);
+            break;
+        case SECURITY_SERVER_BOOTSTRAP_SERVER_ACCOUNT_TIMEOUT_RES_ID:
+            r = sol_lwm2m_tlv_get_int(tlv, &instance_ctx->bootstrap_server_account_timeout);
+            break;
+        default:
+            fprintf(stderr, "tlv type: %u, ID: %" PRIu16 ", Size: %zu, Content: %.*s"
+                " could not be written to Security Server Object at /0/%" PRIu16,
+                tlv->type, tlv->id, tlv->content.used,
+                SOL_STR_SLICE_PRINT(sol_buffer_get_slice(&tlv->content)), instance_id);
+            return -EINVAL;
+        }
+    }
+
+    if (tlvs->len == 1 && r >= 0)
+        printf("TLV written to Security object at /1/%" PRIu16 "/%" PRIu16 "\n",
+            instance_id, tlv->id);
+    else
+        printf("TLV written to Security object at /1/%" PRIu16 "\n", instance_id);
+
+    return r;
+}
+
+static int
+create_security_obj(void *user_data, struct sol_lwm2m_client *client,
+    uint16_t instance_id, void **instance_data,
+    struct sol_lwm2m_payload payload)
+{
+    struct security_obj_instance_ctx *instance_ctx;
+    int r;
+    uint16_t i;
+    struct sol_lwm2m_tlv *tlv;
+
+    if (payload.type != SOL_LWM2M_CONTENT_TYPE_TLV) {
+        fprintf(stderr, "Content type is not in TLV format\n");
+        return -EINVAL;
+    }
+
+    instance_ctx = calloc(1, sizeof(struct security_obj_instance_ctx));
+    if (!instance_ctx) {
+        fprintf(stderr, "Could not alloc memory for security object context\n");
+        return -ENOMEM;
+    }
+
+    SOL_VECTOR_FOREACH_IDX (&payload.payload.tlv_content, tlv, i) {
+        SOL_BUFFER_DECLARE_STATIC(buf, 64);
+
+        if (tlv->id == SECURITY_SERVER_SERVER_URI_RES_ID) {
+            r = sol_lwm2m_tlv_get_bytes(tlv, &buf);
```

Comment:
Changed to
```C
            r = sol_lwm2m_tlv_get_bytes(tlv, &buf);
            if (r < 0)
                goto err_free_instance;
            instance_ctx->server_uri = sol_buffer_to_blob(&buf);
            if (!instance_ctx->server_uri)
                goto err_free_instance;
```

Fields in the other `else if`s are checked by the last `if` already.